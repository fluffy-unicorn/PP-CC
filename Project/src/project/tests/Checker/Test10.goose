# Correct acquire and release on locks
lock l;
shared int sh = 50;

thread a() {
    acquire l;
    print 10;
    print sh;
    print 20;
    release l;
}

function void main() {
    thread a = fork a();
    join a;
}
