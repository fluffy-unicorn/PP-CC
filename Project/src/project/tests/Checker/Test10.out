Sprockell 1 says 10
Sprockell 1 says 50
Sprockell 1 says 20
